Formula/jabba.rb

```diff
@@ -20,25 +21,57 @@ class Jabba < Formula
   depends_on "go" => :build
 
   def install
-    ENV["GOPATH"] = buildpath
-    ENV["GO111MODULE"] = "auto"
-    dir = buildpath/"src/github.com/Jabba-Team/jabba"
-    dir.install buildpath.children
-    cd dir do
-      ldflags = "-X main.version=#{version}"
-      system "go", "build", *std_go_args(ldflags: ldflags)
+    ENV["JABBA_GET"] = "false"
+
+    # Customize install locations
+    # https://github.com/Jabba-Team/jabba/pull/17
+    inreplace "Makefile", " sh install.sh", " bash install.sh --skip-rc"
+    inreplace "install.sh" do |s|
+      s.gsub! "  rm -f", "  command rm -f"
+      s.gsub! "$JABBA_HOME_TO_EXPORT/bin/jabba", "#{opt_bin}/jabba"
+      s.gsub! "${JABBA_HOME}/bin", bin.to_s
+      s.gsub! "${JABBA_HOME}/jabba.sh", "#{pkgshare}/jabba.sh"
+      s.gsub! "${JABBA_HOME}/jabba.fish", "#{pkgshare}/jabba.fish"
     end
+
+    pkgshare.mkpath
+
+    system "make", "VERSION=#{version}", "install"
+  end
+
+  def caveats
+    <<~EOS
+      Add the following line to your ~/.bashrc or ~/.zshrc file:
+        [ -s "#{opt_pkgshare}/jabba.sh" ] && . "#{opt_pkgshare}/jabba.sh"
+
+      If you use the Fish shell then add the following line to your ~/.config/fish/config.fish:
+        [ -s "#{opt_pkgshare}/jabba.fish" ]; and source "#{opt_pkgshare}/jabba.fish"
+    EOS
   end
 
   test do
     jdk_version = "zulu@17"
     version_check ='openjdk version "17'
+    fd3 = Tempfile.open("jabba.fd3")
 
     ENV["JABBA_HOME"] = testpath/"jabba_home"
 
     system bin/"jabba", "install", jdk_version
-    jdk_path = shell_output("#{bin}/jabba which #{jdk_version}").strip
-    jdk_path += "/Contents/Home" if OS.mac?
+    shell_output "#{bin}/jabba use #{jdk_version} 3>#{fd3.path}"
+
+    java_home_pattern = /^export JAVA_HOME="([^"]+)"/
+    jdk_path = ""
+    fd3.rewind
+    fd3.each do |line|
+      m = java_home_pattern.match line
+      next if m.nil?
+
+      jdk_path = m[1]
+    end
+    fd3.unlink
+
+    refute_equal "", jdk_path
```

Comment:
how is all this different than using this?
```rb
assert_match /^export JAVA_HOME="([^"]+)"/, shell_output("#{bin}/jabba use #{jdk_version}")
```

Reply:
It doesn't go to output, so `shell_output` would be empty. It uses a specific file descriptor used by the generated shell script integration to hide the shell commands (setting environment variables) the will be executed in the background. For simplicity sake, I guess we could add `3>&1` to the command to redirect that special stream to the normally empty output of that command.

Reply:
Simplified in latest push.